rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -447,56 +447,26 @@ protected PegTxType getTransactionType(BtcTransaction btcTx) {
         /************************************************************************/
         /** Special case to migrate funds from an old federation               **/
         /************************************************************************/
-        if (activations.isActive(ConsensusRule.RSKIP199) && txIsFromOldFederation(btcTx)) {
+        if (activations.isActive(ConsensusRule.RSKIP199) &&
+            PegUtilsLegacy.txIsFromOldFederation(
+                btcTx,
+                Address.fromBase58(bridgeConstants.getBtcParams(), bridgeConstants.getOldFederationAddress())
+            )
+        ) {
             logger.debug("[getTransactionType][btc tx {}] is from the old federation, treated as a migration", btcTx.getHash());
             return PegTxType.PEGOUT_OR_MIGRATION;
         }
```

Comment:
Shouldn't all this go into `PegUtilsLegacy.getTransactionType` also?

Reply:
It was there, but since your comment above, I moved here because I understood you were referring to the fact that I was mixing oldFederation and retiredFederation logics, so I decided to put it out of the getTransactionType method considering this is a special condition.

Reply:
Sorry for the mix up. But I'd move this to PegUtilsLegacy class yes, inside getTransactionType